src/alert.cpp

```diff
@@ -829,18 +829,67 @@ namespace libtorrent {
 	listen_failed_alert::listen_failed_alert(
 		aux::stack_allocator& alloc
 		, std::string const& iface
-		, tcp::endpoint const& ep
+		, libtorrent::address const& listen_addr
+		, int listen_port
 		, int op
 		, error_code const& ec
 		, socket_type_t t)
 		: error(ec)
 		, operation(op)
 		, sock_type(t)
-		, endpoint(ep)
+		, address(listen_addr)
+		, port(listen_port)
```

Comment:
you need to initialize the endpoint here too, right? (if deprecated features are enabled)